lisp/ess-inf.el

```diff
@@ -718,23 +718,24 @@ LANGUAGE is ignored."
 (defun ess-request-a-process (message &optional noswitch ask-if-1)
   "Ask for a process, and make it the current ESS process.
 If there is exactly one process, only ask if ASK-IF-1 is non-nil.
-Also switches to the process buffer unless NOSWITCH is non-nil.  Interactively,
-NOSWITCH can be set by giving a prefix argument.
-Returns the name of the selected process."
-  (interactive
-   (list "Switch to which ESS process? " current-prefix-arg))
-                                        ; prefix sets 'noswitch
-  (ess-write-to-dribble-buffer "ess-request-a-process: {beginning}\n")
+Also switches to the process buffer unless NOSWITCH is non-nil.
+Interactively, NOSWITCH can be set by giving a prefix argument.
+Returns the name of the selected process. MESSAGE may get passed
+to `ess-completing-read'."
+  (interactive (list "Switch to which ESS process? " current-prefix-arg))
   (update-ess-process-name-list)
-
-  (setq ess-dialect (or ess-dialect
-                        (ess-completing-read
-                         "Set `ess-dialect'"
-                         (delete-dups (list "R" "S+" (or (bound-and-true-p S+-dialect-name) "S+")
-                                            "stata" (or (bound-and-true-p STA-dialect-name) "stata")
-                                            "julia" "SAS" "XLS"  "ViSta")))))
-
-  (let* ((pname-list (delq nil ;; keep only those matching dialect
+  (let* ((ess-dialect (or ess-dialect
+                          (ess-completing-read
+                           "Set `ess-dialect'"
+                           (delete-dups (list "R" "S+" (or (bound-and-true-p S+-dialect-name) "S+")
+                                              "stata" (or (bound-and-true-p STA-dialect-name) "stata")
+                                              "julia" "SAS")))))
+         ;; Set `display-buffer-overriding-action' here since we
+         ;; explicitly handle buffer switching below with
+         ;; `pop-to-buffer' which handles the `display-buffer'
+         ;; framework.
+         (display-buffer-overriding-action '(display-buffer-no-window (allow-no-window . t)))
```

Comment:
So this is the only change right? All others are cleanups of various kind AFAICS. 

If we move this line all the way down to where pop-to-buffer is used, the comment would be no longer necessary.

Reply:
Yes, though this fix isn't complete yet. It's more complicated than I thought and I haven't had time to look at it yet.